translate/automl/src/main/java/com/google/cloud/translate/automl/DatasetApi.java

```diff
@@ -54,53 +54,54 @@
-
-    // Display the dataset information.
-    System.out.println(String.format("Dataset name: %s", dataset.getName()));
-    System.out.println(
-        String.format(
-            "Dataset id: %s",
-            dataset.getName().split("/")[dataset.getName().split("/").length - 1]));
-    System.out.println(String.format("Dataset display name: %s", dataset.getDisplayName()));
-    System.out.println("Translation dataset Metadata:");
-    System.out.println(
-        String.format(
-            "\tSource language code: %s",
-            dataset.getTranslationDatasetMetadata().getSourceLanguageCode()));
-    System.out.println(
-        String.format(
-            "\tTarget language code: %s",
-            dataset.getTranslationDatasetMetadata().getTargetLanguageCode()));
-    System.out.println("Dataset create time:");
-    System.out.println(String.format("\tseconds: %s", dataset.getCreateTime().getSeconds()));
-    System.out.println(String.format("\tnanos: %s", dataset.getCreateTime().getNanos()));
+    try (AutoMlClient client = AutoMlClient.create()) {
+
+      // A resource that represents Google Cloud Platform location.
+      LocationName projectLocation = LocationName.of(projectId, computeRegion);
+
+      // Specify the source and target language.
+      TranslationDatasetMetadata translationDatasetMetadata =
+          TranslationDatasetMetadata.newBuilder()
+              .setSourceLanguageCode(source)
+              .setTargetLanguageCode(target)
+              .build();
+
+      // Set dataset name and dataset metadata.
+      Dataset myDataset =
+          Dataset.newBuilder()
+              .setDisplayName(datasetName)
+              .setTranslationDatasetMetadata(translationDatasetMetadata)
+              .build();
+
+      // Create a dataset with the dataset metadata in the region.
+      Dataset dataset = client.createDataset(projectLocation, myDataset);
+
+      // Display the dataset information.
+      System.out.println(String.format("Dataset name: %s", dataset.getName()));
+      System.out.println(
+          String.format(
+              "Dataset id: %s",
+              dataset.getName().split("/")[dataset.getName().split("/").length - 1]));
+      System.out.println(String.format("Dataset display name: %s", dataset.getDisplayName()));
+      System.out.println("Translation dataset Metadata:");
+      System.out.println(
+          String.format(
+              "\tSource language code: %s",
+              dataset.getTranslationDatasetMetadata().getSourceLanguageCode()));
+      System.out.println(
+          String.format(
+              "\tTarget language code: %s",
+              dataset.getTranslationDatasetMetadata().getTargetLanguageCode()));
+      System.out.println("Dataset create time:");
+      System.out.println(String.format("\tseconds: %s", dataset.getCreateTime().getSeconds()));
+      System.out.println(String.format("\tnanos: %s", dataset.getCreateTime().getNanos()));
+    } catch (IOException e) {
```

Comment:
In our maintenance meeting the other day, we discussed catching exceptions in this way.

The consensus is: for exceptions, if there is some specific advice we can give the user to handle it, then we should include that at least in a comment.

Otherwise, the exception should pass through the function with `throws`.

Please rework this part (and all other places where you removed a `throws` from the function heading) so that it fits one of those two.